allennlp/commands/elmo.py

```diff
@@ -0,0 +1,203 @@
+import torch
+
+from allennlp.common.tqdm import Tqdm
+from allennlp.common.util import lazy_groups_of
+from allennlp.data.dataset import Batch
+from allennlp.data import Token, Vocabulary, Instance
+from allennlp.data.fields import TextField
+from allennlp.data.token_indexers.elmo_indexer import ELMoTokenCharactersIndexer
+from allennlp.nn.util import remove_sentence_boundaries
+from allennlp.modules.elmo import _ElmoBiLm
+from allennlp.commands.subcommand import Subcommand
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+DEFAULT_OPTIONS_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_options.json" # pylint: disable=line-too-long
+DEFAULT_WEIGHT_FILE = "https://s3-us-west-2.amazonaws.com/allennlp/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_weights.hdf5" # pylint: disable=line-too-long
+
+
+class Elmo(Subcommand):
+    def add_subparser(self, name: str, parser: argparse._SubParsersAction) -> argparse.ArgumentParser:
+        # pylint: disable=protected-access
+        description = '''Create word vectors using ELMo.'''
+        subparser = parser.add_parser(
+                name, description=description, help='Use a trained model to make predictions.')
+
+        subparser.add_argument('input_file', type=argparse.FileType('r'), help='path to input file')
+        subparser.add_argument('output_file', type=str, help='path to output file')
+
+        subparser.add_argument('--vocab-path', type=str, help='A path to a vocabulary file to generate ')
+        subparser.add_argument(
+                '--options-file',
+                type=str,
+                default=DEFAULT_OPTIONS_FILE,
+                help='The path to the ELMo options file.')
+        subparser.add_argument(
+                '--weight-file',
+                type=str,
+                default=DEFAULT_WEIGHT_FILE,
+                help='The path to the ELMo weight file.')
+        subparser.add_argument('--batch-size', type=int, default=64, help='The batch size to use.')
+        subparser.add_argument('--cuda-device', type=int, default=-1, help='The cuda_device to run on.')
+        subparser.add_argument('--use-sentence-key', default=False, action='store_true')
+
+        subparser.set_defaults(func=elmo_command)
+
+        return subparser
+
+
+class ElmoEmbedder():
+    def __init__(self,
+                 options_file: str = DEFAULT_OPTIONS_FILE,
+                 weight_file: str = DEFAULT_WEIGHT_FILE,
+                 cuda_device: int = -1) -> None:
+        self.indexer = ELMoTokenCharactersIndexer()
+
+        logger.info("Initializing ELMo.")
+        self.elmo_bilm = _ElmoBiLm(options_file, weight_file)
+        if cuda_device >= 0:
+            self.elmo_bilm = self.elmo_bilm.cuda(cuda_device=cuda_device)
+
+        self.cuda_device = cuda_device
+
+    def batch_to_ids(self, batch: List[List[str]]) -> torch.Tensor:
+        """
```

Comment:
Can you use the Sphinx docstrings throughout this class? It needs to be:

``` 
Parameters
-----------
name : type
    description

Returns
--------

....
```

Reply:
Oops--sorry I had PyCharm create these and didn't realize it was different.  Will do.

Reply:
Done.